flashlight/fl/tensor/backend/jit/eval/Evaluator.cpp

```diff
@@ -52,16 +52,45 @@ void Evaluator::evalCustomNode(CustomNode& node) {
 }
 
 void Evaluator::evalIndexNode(IndexNode& node) {
-  std::vector<Index> indices;
-  for (const auto& index : node.indices()) {
+  const auto& indexedTensor = node.indexedNode()->getResult().value();
+  node.setResult(indexedTensor(unwrapTensorInIndices(node.indices())));
+}
+
+void Evaluator::evalIndexedUpdateNode(IndexedUpdateNode& node) {
+  // TODO no need to copy if indexedNode has only 1 user here
```

Comment:
Why not implement it? Seems like a very useful optimization. Feel free to add in a downstream PR too